src/node_file.cc

```diff
@@ -102,34 +103,30 @@ using v8::Value;
 
 #define GET_OFFSET(a) ((a)->IsNumber() ? (a)->IntegerValue() : -1)
 
-FSReqWrap* FSReqWrap::New(Environment* env,
-                          Local<Object> req,
-                          const char* syscall,
-                          const char* data,
-                          enum encoding encoding,
-                          Ownership ownership) {
-  const bool copy = (data != nullptr && ownership == COPY);
-  const size_t size = copy ? 1 + strlen(data) : 0;
-  FSReqWrap* that;
-  char* const storage = new char[sizeof(*that) + size];
-  that = new(storage) FSReqWrap(env, req, syscall, data, encoding);
-  if (copy)
-    that->data_ = static_cast<char*>(memcpy(that->inline_data(), data, size));
-  return that;
-}
-
-
 void FSReqWrap::Dispose() {
+  if (info_ != nullptr)
+    info_->Dispose();
   this->~FSReqWrap();
```

Comment:
Leaks `this`, i.e., the `FSReqWrap`.  Should probably be just `delete wrap` at the call site and move the `if (info_ != nullptr)` bit to the destructor (and remove this method because it's no longer used after that.)